vimeo-networking/src/main/java/com/vimeo/networking/logging/ClientLogger.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Vimeo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.vimeo.networking.logging;
+
+import com.vimeo.networking.VimeoClient;
+
+/**
+ * Simple utility class to delegate logging when it is not known whether VimeoClient has been initialized yet
+ * <p/>
+ * Created by zetterstromk on 6/15/16.
+ */
+public final class ClientLogger {
```

Comment:
When wouldn't VimeoClient be initialized yet? It now seems like we have two different routes we could go through to log something in VimeoClient which feels like it might be too much (like too many classes doing the same thing). Because now we have ClientLogger and DebugLogger which both handle logs - and then this class and VimeoClient which both hold and instance of a logger. Additionally, since this is a wrapper anytime we add something to the DebugLoggerInterface, we'll also have to add it to our DebugLogger and then we'll also have to add it to this class. What if we had a static method on VimeoClient that returned a DebugLoggerInterface - that removes convenience but minimizes the needs to make modifications. Or maybe making DebugLogger.java static kind of combining this class and DebugLogger.  Or maybe ensuring that VimeoClient is actually initialized before hand.
I'm just spitballing, it just seems like we have a bunch of classes and instances around logging so if we can minimize the logging footprint, it would probably be beneficial.